src/groupByKey.ts

```diff
@@ -2,6 +2,20 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<R>(
+  items: R[],
+  key: keyof R,
+): GroupsMap<R> {
+  return items.reduce((acc: GroupsMap<R>, item: R) => {
+    const groupKey:string = `${item[key]}`;
+    const groups = { ...acc };
+
+    if (groups[groupKey]) {
+      groups[groupKey].push(item);
+    } else {
+      groups[groupKey] = [item];
+    }
```

Comment:
i believe you could try something like this:
```suggestion
      groups[groupKey] = (groups[groupKey] || []).concat(item);
```